src/librbd/MirroringWatcher.cc

```diff
@@ -108,8 +108,8 @@ void MirroringWatcher<I>::handle_notify(uint64_t notify_id, uint64_t handle,
     return;
   }
 
-  apply_visitor(watcher::util::HandlePayloadVisitor<MirroringWatcher<I>>(
-                  this, notify_id, handle), notify_message.payload);
+  visit(watcher::util::HandlePayloadVisitor<MirroringWatcher<I>>(
+	  this, notify_id, handle), notify_message.payload);
```

Comment:
It's better qualify the call -- I hope we get rid of remaining `using namespace std` instances in random header files one day:
```suggestion
  std::visit(watcher::util::HandlePayloadVisitor<MirroringWatcher<I>>(
	       this, notify_id, handle), notify_message.payload);
```
This applies throughout the PR.

Reply:
i think that this is how ADL works, and we are not relying `using namespace std` in this case.  see https://gcc.gnu.org/bugzilla/show_bug.cgi?id=82619

Reply:
> It's better qualify the call -- I hope we get rid of remaining `using namespace std` instances in random header files one day:
> 

we've removed a lot of `using namespace std` in header files before. see #42742 and #44736. and the only leftover is:
```console
$ git grep 'using namespace std;' | grep '\.h'
src/tools/ceph_dedup/common.h:using namespace std;
```
#63145 was created to address it.

Reply:
fair enough. i intended to create a minimal change which only addresses the `boost::variant` migration instead of changing whatever not looks nice. but if qualifying the `visit()` with `std::` helps, i am not against including the change in my PR. will do it.